RadarSDK/RadarLocationManager.m

```diff
@@ -590,19 +586,12 @@ - (void)replaceSyncedGeofences:(NSArray<RadarGeofence *> *)geofences {
         }
     }
 
-    [self removePendingNotificationsWithCompletionHandler: ^{
-        for (UNNotificationRequest *request in requests) {
-            [self.notificationCenter addNotificationRequest:request withCompletionHandler:^(NSError *_Nullable error) {
-                if (error) {
-                    [[RadarLogger sharedInstance]
-                        logWithLevel:RadarLogLevelDebug
-                            message:[NSString stringWithFormat:@"Error adding local notification | identifier = %@; error = %@", request.identifier, error]];
-                } else {
-                    [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelDebug
-                                                        message:[NSString stringWithFormat:@"Added local notification | identifier = %@", request.identifier]];
-                }
-            }];
-        }
+    [RadarNotificationHelper checkForSentOnPremiseNotifications];
+
+    [RadarState clearPendingNotificationRequests];
+    
+    [RadarNotificationHelper removePendingNotificationsWithCompletionHandler: ^{
+        [RadarNotificationHelper addOnPremiseNotificationRequests:requests];
```

Comment:
Possible race? Do we need to call `removePendingNotificationsWithCompletionHandler` in a completion handler of `clearPendingNotificationRequests` (which itself maybe is called in a completion handler of `checkForSentOnPremiseNotifications`)

Reply:
we do indeed need a completion handler for `checkForSentOnPremiseNotifications ` but not for `clearPendingNotificationRequests` since it does not step on the other calls.